e2e/e2e_test.go

```diff
@@ -0,0 +1,168 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/utils/pointer"
+)
+
+const (
+	blockingPodName = "api-blocker-pod"
+	safeToAssumeNodeRebootTimeout = 180 * time.Second
+	testNamespace = "default"
+)
+
+var _ = Describe("e2e", func() {
+	var targetNode string
+
+	BeforeEach(func() {
+		// randomly pick a host (or let the scheduler do it by running the blocking pod)
+		// block the api port to make it go Ready Unknown
+		node, err := makeNodeUnready(time.Minute * 5)
+		Expect(err).NotTo(HaveOccurred())
+		targetNode = node
+	})
+
+	AfterEach(func() {
+		removeAPIBlockingPod()
+	})
+	When("Node conditions meets the unhealthy criteria", func() {
+		It("Remediates a host", func() {
+			Expect(nhcRemediatesTheHost(targetNode, 2*time.Minute)).To(Succeed())
+			Expect(nodeObjectIsRemoved(targetNode, safeToAssumeNodeRebootTimeout+30*time.Second)).To(Succeed())
+		})
+	})
+})
+
+func nodeObjectIsRemoved(nodeName string, timeout time.Duration) error {
+	fmt.Fprintf(GinkgoWriter, "Node %v should be removed by ppill. Waiting for %v\n", nodeName, timeout)
+	err := wait.Poll(5*time.Second, timeout, func() (done bool, err error) {
+		_, err = clientSet.CoreV1().Nodes().
+			Get(context.Background(),
+				nodeName,
+				metav1.GetOptions{})
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				fmt.Fprintf(GinkgoWriter, "Node %v is not found and probably was removed by ppill\n", nodeName)
+				return true, nil
+			}
+			return false, err
+		}
+		fmt.Fprintf(GinkgoWriter, "node %v still found and was not removed", nodeName)
+		return false, nil
+	})
+	return err
+}
+
+// nhcRemediatesTheHost tracks the ppil object with a name like the node, and returns
+// when it finds one, or timeouts after duration
+func nhcRemediatesTheHost(nodeName string, timeout time.Duration) error {
+	fmt.Fprintf(GinkgoWriter, "Node %v should be picked up by NHC and Poison-Pill. Waiting for %v\n", nodeName, timeout)
+	// wait till we see a remediation cr with the node name
+	err := wait.Poll(5*time.Second, timeout, func() (done bool, err error) {
```

Comment:
instead of `wait.Poll` you might want to consider using Ginkgo functions like `Eventually`

Reply:
example:
https://github.com/medik8s/poison-pill/blob/master/controllers/controller_test.go#L130-L138

Reply:
Its more elegant indeed. I'll use it

Reply:
done